app/models/king.rb

```diff
@@ -59,19 +64,25 @@ def initial_state?(rook)
 
   def puts_in_check?(rook)
     if queenside?(rook) # rook on queenside
-      first_y_space = -1 # move left one
-      second_y_space = -2 # move left two
+      first_x_space = -1 # move left one
+      #second_x_space = -2 # move left two
```

Comment:
also fixed up the castling method on the backend to actually implement check correctly